go/tasks/pluginmachinery/k8s/config.go

```diff
@@ -0,0 +1,58 @@
+package k8s
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	restclient "k8s.io/client-go/rest"
+)
+
+type ClusterConfig struct {
+	Name     string `json:"name" pflag:",Friendly name of the remote cluster"`
+	Endpoint string `json:"endpoint" pflag:", Remote K8s cluster endpoint"`
+	Auth     Auth   `json:"auth" pflag:"-, Auth setting for the cluster"`
+	Enabled  bool   `json:"enabled" pflag:", Boolean flag to enable or disable"`
+}
+
+type Auth struct {
+	Type      string `json:"type" pflag:", Authentication type"`
+	TokenPath string `json:"token-path" pflag:", Token path"`
+	CertPath  string `json:"cert-path" pflag:", Certificate path"`
+}
+
+func (auth *Auth) GetCA() ([]byte, error) {
+	cert, err := ioutil.ReadFile(auth.CertPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read k8s CA cert from configured path")
+	}
+	return cert, nil
+}
+
+func (auth *Auth) GetToken() (string, error) {
+	token, err := ioutil.ReadFile(auth.TokenPath)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to read k8s bearer token from configured path")
+	}
+	return string(token), nil
+}
+
+func NewK8sClusterConfig(host string, auth Auth) (*restclient.Config, error) {
+	tokenString, err := auth.GetToken()
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Failed to get auth token: %+v", err))
+	}
+
+	caCert, err := auth.GetCA()
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Failed to get auth CA: %+v", err))
+	}
+
+	tlsClientConfig := restclient.TLSClientConfig{}
```

Comment:
is NOT providing a CA cert a valid config? if so, should you or should you not create a TLS client config? should that be an option `InsecureConnection bool` ?

Reply:
I believe we can. I was keeping it simple since users supplying this config will have the token and ca cert at their disposal. This kinda avoids users making mistakes as insecure connection should only be used for testing.

Should we go ahead and add it anyway?

Reply:
If you are going over the internet here... like exposing ApiServer through LoadBalancer... etc. etc. you will probably not use the self-signed CA cert and use a CloudProvider-provided Cert... will that still work? I imagine it's common to terminate SSL at ingress point in the remote cluster and let traffic be unauthenticated after... or maybe not... 

I'm always wary (as you can tell) when it comes to security, so I really appreciate you going through the exercise to figure out the right way to do this...

Another option is to just expose the ClientConfig in our config object and let people specify all the params they need... not sure how bad that can be...

Reply:
Is not clear to me yet what other usages are sensible/common for these cases. That said, one thing we can do is just drop this config and have `NewKubeClient` have the `restclient.Config`.

The downside is that plugin creators might create different remote cluster configs according to their use case.

Reply:
>The downside is that plugin creators might create different remote cluster configs according to their use case.

actually, when it comes to the usage of this in plugins it will be up to the user to define the `restclient.Config` for their use case (since they will supply the func that actually creates the KubeClient). So having `restclient.Config` being passed to `NewKubeClient` is perfectly fine because 1) this is a very advanced thing 2) setups can vary a lot.

Reply:
Fair... for plugin developers, they can add restclient.Config as a config option and it should be parsed correctly so they can just do `NewKubeClient(GetConfig().KubeClient)` or something...